app/src/main/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentService.java

```diff
@@ -235,13 +247,48 @@ private JsonObject customContentTab(String tabType, String name, JsonObject prop
         return result;
     }
 
-    // Smart-Seq-like experiments will contain the substring “smart” in their technology types
-    private static boolean isSmartExperiment(Collection<String> technologyType) {
-        return technologyType.stream()
-                .anyMatch(type -> type.toLowerCase().matches("smart" + "-(?:.*)"));
+    public ImmutableMap fetchDefaultPlotMethodAndParameterisation(String experimentAccession) {
+        return cellPlotService.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
     }
 
-    public ImmutableMap fetchDefaultPlotMethodAndParameterisation(String experimentAccession){
-        return cellPlotService.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
+    public JsonArray getMarkerGeneMetadata(JsonArray metadata, String experimentAccession) {
+        JsonArray markerGenesArray = new JsonArray();
+
+        metadata.forEach(item -> {
+            JsonObject jsonObject = item.getAsJsonObject();
+            processMarkerGene(jsonObject, experimentAccession,
+                    "inferred_cell_type_-_ontology_labels",
+                    "inferred cell type - ontology labels",
+                    markerGenesArray);
+            processMarkerGene(jsonObject, experimentAccession,
+                    "inferred_cell_type_-_authors_labels",
+                    "Inferred cell type - authors labels",
+                    markerGenesArray);
+        });
+
+        return markerGenesArray;
+    }
+
+    private void processMarkerGene(JsonObject jsonObject, String experimentAccession,
+                                   String keyValue, String serviceLabel, JsonArray targetArray) {
+        if (containsValue(jsonObject, "value", keyValue)) {
+            boolean isAvailable = markerGeneService.isMarkerGenesAvailableForTheInferredCellTypes(
+                    experimentAccession, serviceLabel) > 0;
+            jsonObject.addProperty("status", Boolean.toString(isAvailable));
+            targetArray.add(jsonObject);
+        }
+    }
+
+    private boolean containsValue(JsonObject jsonObject, String key, Object value) {
+        if (jsonObject.has(key)) {
+            try {
+                String jsonValue = jsonObject.get(key).getAsString();
+                return value != null && value.toString().equals(jsonValue);
```

Comment:
The `value` variable never `null` in your usage. I checked it in IntelliJ and I am not sure why it is not highlighted to you this fact. Anyway, we don't need to check if it is `null`.
Could you remove that, please?
Thanks